views/databaseIssue.ejs

```diff
@@ -0,0 +1,250 @@
+                        <!--                        </small>-->
+                        <!--                        <h3 class="badge badge-info">-->
+                        <!--                            mongodb://username:password@host:port/octofarm?authSource=admin-->
+                        <!--                        </h3>-->
+                        <!--                        <small id="authSourceHelp" class="form-text text-muted">-->
+                        <!--                            For authentication the authSource parameter might be required. Add the-->
+                        <!--                            following in that case:-->
+                        <!--                        </small>-->
+                        <!--                        <h3 class="badge badge-info">?authSource=admin</h3>-->
+                    </div>
+                    <br />
+                    <button type="button"
+                            id="test-connection-button"
+                            class="btn btn-warning"
+                            onclick="postTestConnection()"
+                            disabled>
+                        Test connection
+                    </button>
+                    <button
+                            type="button"
+                            id="save-connection-button"
+                            class="btn btn-success"
+                            onclick="postSaveConnection()"
+                            <% if (isDocker) { %>
+                            hidden
+                            <% } %>
+                            disabled>
+                        Save connection setting
+                    </button>
+                    <% if (!isDocker) { %>
+                    <small id="not-yet-validated-connection" class="form-text text-warning">
+                        Test your connection before saving.
+                    </small>
+                    <% } else { %>
+                        <div class="text-warning">
+                            Running OctoFarm in a docker container. Update the MONGO variable in your docker command `-e MONGO=...` or docker-compose.yml file's `environment:` section.
+                            <br/>
+                            You can still use the test button above. Make sure to restart the container after you've managed to establish a successful connection here.
+                        </div>
+                    <% } %>
+
+                    <div class="list-group" style="margin-top: 15px">
+                        Last OctoFarm response:
+                        <small id="connection-string-server-response" class="text-light">
+                            not requested yet
+                        </small>
+                    </div>
+                </form>
+            </div>
+        </div>
+    </div>
+</div>
+
+<script>
```

Comment:
Inline javascript so we never have to compile: I find it nice to be sure that the fallback is self-sufficient